### PR TITLE
Update Rename UI to accommodate for getting suggestions automatically

### PR DESCRIPTION
**Pull request by AmadeusW**
We're introducing an experiment to automatically request Rename suggestions.
We don't want to automatically obstruct the Rename options with suggested names, so we've worked with PMs and designers to come up with a UI that always accommodates for 3 results, and gets them automatically.
User may opt out of this behavior by toggling the Copilot button (via mouse or hitting Ctrl+Space), which would restore the default UI.

### Inline rename with Copilot, suggestions are obtained automatically
_✨ this is the new scenario_
roslyn NEW behavior

Ctrl+Space works to toggle whether suggestions are obtained automatically. If they were off, hitting Ctrl+Space will request suggestions. This state is saved for next time when the rename dialog opens.
roslyn NEW persistence

String changes:
- Removed the Shift+Space comment, which drastically narrowed the window back to its original size
- Adjusted the tooltip (this is a new string and may need to be localized)
image

### For reference: Inline rename with Copilot, existing behavior 
where Ctrl+Space or button press gets suggestions
Here, the copilot button is a bit smaller (and chevron is larger) so that buttons line up. Additionally, the progress bar does not have a light gray background that it had before.
roslyn NEW manual smart rename

### For reference: Inline rename without Copilot 
Here, the chevron button is a bit larger. I had to make this change so that the buttons line up in all 3 scenarios. This increased height of the textbox by 2px, which I believe looks better.
roslyn NEW no smart rename2

### CODE REVIEW

**Comment by Cosifne** [comment]
Do you want to target release/dev17.10?

**Comment by AmadeusW** [comment]
> Do you want to target release/dev17.10?

yes, thank you!

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs [hunk 1]
```diff
@@ -350,6 +350,8 @@ public bool TryFetch(LocalUserRegistryOptionPersister persister, OptionKey2 opti
         {"dotnet_collapse_inline_rename_ui", new RoamingProfileStorage("TextEditor.CollapseRenameUI")},
         {"dotnet_rename_use_inline_adornment", new RoamingProfileStorage("TextEditor.RenameUseInlineAdornment")},
         {"dotnet_preview_inline_rename_changes", new RoamingProfileStorage("TextEditor.Specific.PreviewRename")},
+        {"dotnet_collapse_suggestions_in_inline_rename_ui", new RoamingProfileStorage("TextEditor.CollapseRenameSuggestionsUI")},
+        {"dotnet_rename_get_suggestions_automatically", new FeatureFlagStorage(@"Editor.AutoSmartRenameSuggestions")},
```
Comment (AmadeusW) [inline src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:354]: The feature flag is defined as follows in the .pkgdef file:
```
[$RootKey$\FeatureFlags\Editor\AutoSmartRenameSuggestions]
"Value"=dword:00000000
```

File: src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml [hunk 2]
```diff
@@ -202,15 +210,67 @@
...
+                            Background="{DynamicResource {x:Static vsui:EnvironmentColors.ComboBoxPopupBackgroundGradientBrushKey}}">
+                            <StackPanel x:Name="PanelProgressBarHost"
+                                        x:Uid="PanelProgressBarHost"
+                                        Orientation="Vertical"
+                                        Visibility="{Binding IsInProgress, Converter={StaticResource BooleanToHiddenVisibilityConverter}, Mode=OneWay}"
+                                        >
+                                <TextBlock x:Name="PanelLabel"
+                                           x:Uid="PanelLabel"
+                                           Padding="2"
+                                           Text="Generating suggestions..." />
```
Comment (Cosifne) [inline src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml:254]: Does this need LOC?

File: src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs [hunk 3]
```diff
@@ -85,11 +120,24 @@ internal sealed class SmartRenameViewModel : INotifyPropertyChanged, IDisposable
         this.BaseViewModel.IdentifierText = baseViewModel.IdentifierText;
 
         GetSuggestionsCommand = new DelegateCommand(OnGetSuggestionsCommandExecute, null, threadingContext.JoinableTaskFactory);
+
+        var getSuggestionsAutomatically = _globalOptionService.GetOption(InlineRenameUIOptionsStorage.GetSuggestionsAutomatically);
```
Comment (Cosifne) [inline src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs:125]: should `InlineRenameUIOptionsStorage.GetSuggestionsAutomatically` be called `nlineRenameUIOptionsStorage.UseSuggestionsPanel`?  I was confused when looking at this.

Comment (Cosifne) [reply src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs:127]: Also `InlineRenameUIOptionsStorage.CollapseSuggestionsPanel` seems like controlling the `GetSuggestionsAutomatically` behavior. So I guess they might need a better name

File: src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs [hunk 4]
```diff
@@ -85,11 +120,24 @@ internal sealed class SmartRenameViewModel : INotifyPropertyChanged, IDisposable
...
+        IsUsingResultPanel = getSuggestionsAutomatically;
+        IsUsingDropdown = !IsUsingResultPanel;
```
Comment (Cosifne) [inline src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs:129]: Maybe just use one bool value, or enum. I feel tracking two mutual exclusive states are quite confusing..

File: src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel_Telemetry.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
+﻿using System.Diagnostics.CodeAnalysis;
```
Comment (Cosifne) [inline src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel_Telemetry.cs:1]: Ah I made the mistake, it should contain the lisense header

**Comment by Cosifne** [comment]
Overall lgtm, we might want to
1. See if the telemetry meets PM's needs, and test it works locally.
2. Validate that PR in 17.10. Then find QB next week.


cc @dibarbet  this is another QB PR.

**Comment by AmadeusW** [comment]
Per feedback, I updated the layout and color of the button

Dark theme, collapsed
image

Dark theme, expanded
image

Blue theme, collapsed
image

Blue theme, expanded
image

Light theme, collapsed
image

Light theme, expanded
image

**Review by Cosifne — APPROVED** [review]

**Pull request merged**